### PR TITLE
feat: use formbricks/api package internally in /js & /lib

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
Instead of calling the client APIs ie `response` & `display`, we use the `@formbricks/api` package instead to call them.

## Type of change
- [x] Chore (refactoring code, technical debt, workflow improvements)

### Required
- [ ] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/lib/client/display.ts [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-    console.error(res.text);
+    console.error(res.error);
     throw new Error("Could not create display");
   }
-  const resJson = await res.json();
-  return resJson.data;
+  return res.data;
 };
 
 export const updateDisplay = async (displayId: string, displayInput: any, apiHost: string): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (mattinannt) [reply packages/lib/client/display.ts:21]: Just rethought the approach. What I would like even more is that we don't need the `packages/lib/displays` & `packages/lib/responses` files and use the api directly in `responseQueue`, `widget.ts` & `LinkSurvey.tsx` :-)

File: apps/web/app/s/[surveyId]/components/LinkSurvey.tsx [hunk 2]
```diff
@@ -137,7 +137,18 @@ export default function LinkSurvey({
           formbricksSignature={product.formbricksSignature}
           onDisplay={async () => {
             if (!isPreview) {
-              const { id } = await createDisplay({ surveyId: survey.id }, webAppUrl);
+              const api = new FormbricksAPI({
+                apiHost: webAppUrl,
+                environmentId: "",
```
Comment (mattinannt) [inline apps/web/app/s/[surveyId]/components/LinkSurvey.tsx:142]: @ShubhamPalriwala Not setting the environmentId looks very dangerous. Please make the environmentId mandatory and always set it :-)

File: packages/api/src/api/client/display.ts [hunk 3]
```diff
@@ -20,4 +20,8 @@ export class DisplayAPI {
   async markResponded({ displayId }: { displayId: string }): Promise<Result<TDisplay, NetworkError | Error>> {
```
[the remark this answers is not in the document]
Comment (mattinannt) [reply packages/api/src/api/client/display.ts:20]: Above there is a `markDisplayedForPerson({` method. We only have two methods in displays:
- Create (POST) (on /displays)
- Update (PUT)  (on /displays/[displayId])

The API package should reflect those endpoints.

File: packages/js/src/lib/widget.ts [hunk 4]
```diff
@@ -59,13 +59,19 @@ export const renderWidget = (survey: TSurveyWithTriggers) => {
...
-        const { id } = await createDisplay(
-          {
-            surveyId: survey.id,
-            personId: config.get().state.person.id,
-          },
-          config.get().apiHost
-        );
+        const api = new FormbricksAPI({
+          apiHost: config.get().apiHost,
+          environmentId: "",
```
Comment (mattinannt) [inline packages/js/src/lib/widget.ts:64]: please set environmentId

**Comment by mattinannt** [comment]
@ShubhamPalriwala Thanks for the latest changes. I left a few more comments as I think the api package isn't really intuitive yet.

**Comment by ShubhamPalriwala** [comment]
Hey @mattinannt just made the changes as you suggested!

File: packages/api/src/api/client/display.ts
Comment (mattinannt) [inline packages/api/src/api/client/display.ts:1]:
```ts
export const ZDisplayCreateInput = z.object({
  surveyId: z.string().cuid2(),
  personId: z.string().cuid2().optional(),
  responseId: z.string().cuid2().optional(),
});

export type TDisplayCreateInput = z.infer<typeof ZDisplayCreateInput>;

export const ZDisplayUpdateInput = z.object({
  surveyId: z.string().cuid2(),
  personId: z.string().cuid2().optional(),
  responseId: z.string().cuid2().optional(),
});

export const ZDisplayUpdate = z.object({
  responseId: z.string().cuid2().optional(),
});

export type TDisplayUpdateInput = z.infer<typeof ZDisplayUpdateInput>;
```

**Comment by ShubhamPalriwala** [comment]
Used the new types 🚀

**Comment by mattinannt** [comment]
Currently still testing this before merging and something is off: I can't see the display counting up when I open the link survey; it's only showing it after I responded. I see the display network request but it's not registered in the survey. Still need to determine if this is an issue with this PR or an issue on main.

**Pull request merged**